bin/varnishd/cache/cache_req_body.c

```diff
@@ -224,12 +258,17 @@ ssize_t
 VRB_Iterate(struct worker *wrk, struct vsl_log *vsl,
     struct req *req, objiterate_f *func, void *priv)
 {
+	unsigned allow_partial;
 	int i;
 
+	CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);
 	CHECK_OBJ_NOTNULL(req, REQ_MAGIC);
+	AN(vsl);
 	AN(func);
 
-	if (req->req_body_cached) {
+	allow_partial = (req->vsl == vsl) && (func != httpq_req_body_discard);
```

Comment:
I am not sure about these semantics.
Sure we want to blacklist `httpq_req_body_discard`, but other than that the caller should be in control if they want to process the body partially or fully.
But maybe this is for another PR...

Reply:
Different VSL simply means different thread in this context. If you are in a backend task you can only consume the entire body.